src/object/cli_ec.c

```diff
@@ -2639,7 +2636,74 @@ obj_recx_ec2_daos(struct daos_oclass_attr *oca, int shard,
 		}
 	}
 
-	*output_nr = total;
-	*output_recxs = tgt_recxs;
+	D_FREE(*recxs_p);
+	*recxs_p = tgt_recxs;
+	*nr = total;
+	return 0;
+}
+
+/* Convert DAOS offset to specific data target daos offset */
+int
+obj_recx_ec_daos2shard(struct daos_oclass_attr *oca, int shard,
+		       daos_recx_t **recxs_p, unsigned int *iod_nr)
+{
+	daos_recx_t	*recx = *recxs_p;
+	int		nr = *iod_nr;
+	int		cell_nr = obj_ec_cell_rec_nr(oca);
+	int		stripe_nr = obj_ec_stripe_rec_nr(oca);
+	daos_recx_t	*tgt_recxs;
+	int		total;
+	int		idx;
+	int		i;
+
+	D_ASSERT(shard % obj_ec_tgt_nr(oca) < obj_ec_data_tgt_nr(oca));
+	for (i = 0, total = 0; i < nr; i++) {
+		uint64_t offset = recx[i].rx_idx;
+		uint64_t end = offset + recx[i].rx_nr;
+
+		while (offset < end) {
+			int tgt_idx = obj_ec_tgt_of_recx_idx(offset,
+						stripe_nr, cell_nr);
+			if (tgt_idx <= shard)
```

Comment:
for above "shard", should use "shard % obj_ec_tgt_nr(oca)" instead?

Reply:
ah, right.